sdks/java/core/src/test/java/org/apache/beam/sdk/coders/CommonCoderTest.java

```diff
@@ -0,0 +1,194 @@
+package org.apache.beam.sdk.coders;
+
+import static com.google.common.base.MoreObjects.firstNonNull;
+import static com.google.common.base.Preconditions.checkNotNull;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasItem;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.io.CharStreams;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+import org.apache.beam.sdk.coders.Coder.Context;
+import org.apache.beam.sdk.util.CoderUtils;
+import org.apache.beam.sdk.values.KV;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
+
+/**
+ * Tests that Java SDK coders standardized by the Fn API meet the common spec.
+ */
+@RunWith(Parameterized.class)
+public class CommonCoderTest {
+  private static final String STANDARD_CODERS_YAML_PATH =
+      "/org/apache/beam/fn/v1/standard_coders.yaml";
+
+  private static final Map<String, Class<?>> coders = ImmutableMap.<String, Class<?>>builder()
```

Comment:
It was not obvious to me at this time that anything in this file is useful outside this test. Specifically, all of the classes in this file and how they are instantiated are based on a made-up-spec defined only in the test yaml. So I did not yet think there was any core SDK code to be used.

Once the runner API ("common API"?) is checked into the repository, I could expect us to move the `coder` field in the yaml to that format.